### DISCUSSION TITLE
Sharing scripts between Nuxt 3 projects.

### DISCUSSION THREAD

**Discussion by XPAULUSX** [nuxt/nuxt#16127] [Questions] [answered]
Hello, 

For the purposes of one of my projects, i'd like to be able to share common functionality between multiple nuxt 3 projects.

The functionality i'd like to share will be in the form of a Vue 3 composable that makes use of various Nuxt based composables including useCookie and useRequestHeaders.

The idea is that this file/package could be added to both nuxt projects as an npm package and imported and used in the same way for each.

Obviously, this npm package would need some sort of reference to useCookie and useRequestHeaders, but i'm just not sure where these need to be imported from (as these just work in nuxt out of the box, but obviously will be in a completely separate location at this point).

Hope this makes sense, any help appreciated.

Best,

Paul

**Comment by tylerforesthauser** [comment, marked as answer]
Seems to me you're looking for [Layers](https://nuxt.com/docs/guide/going-further/layers)

**Reply by XPAULUSX** [reply]
This is perfect, exactly what I needed and so straight forward to implement!  Thank you :)
